experiment/process-e2e-suite-log/fileSplitter.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+type FileElement struct {
+	file   *os.File
+	writer *bufio.Writer
+}
+
+// This returns a function that will return a fileElement. This fileElement will be referencing a file
+// whose name will respond to an expression like <prefix>_<counter>.
+// "counter" will increase on each invocation.
+func newFileElementSequencer(prefixToUse string) func() (*FileElement, error) {
```

Comment:
Could we use [`ioutil.TempFile()`](https://golang.org/pkg/io/ioutil/#TempFile)?

Reply:
Thanks for showing me this utility :) . Explored it and learnt. Thanks.

For this section of code tried to mimic what [csplit](https://www.gnu.org/software/coreutils/manual/html_node/csplit-invocation.html) does (initially used that in the bash script), and I found it useful that the grammar of the files being generated responded to an monotonic fashion that helped me knowing with file came first. Exploring TempFile the "digit" section of the name formation  it does not collide but is not monotonic. The intention down the road was to have csplit as an inspiration. Let  me know your opinion on this :)